### PR TITLE
feat(bug) : Releases attachment download issue fix

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue:  #2136 

Using postman. 

URL : http://localhost:8080/resource/api/releases/caf6ff2180064d9faaebe8d7a7cd5ae2/attachments

Sample Response in the following attachment : 

image

> Have you implemented any additional tests?
No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.

image

**Comment by nikkuma7** [comment]
Test successful.

image

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/ReleaseController.java [hunk 1]
```diff
@@ -486,11 +486,11 @@ public ResponseEntity<EntityModel<Release>> createRelease(
     }
 
     @GetMapping(value = RELEASES_URL + "/{id}/attachments")
-    public ResponseEntity<CollectionModel<EntityModel<AttachmentDTO>>> getReleaseAttachment1s(
+    public ResponseEntity<CollectionModel<EntityModel<Attachment>>> getReleaseAttachments(
             @PathVariable("id") String id) throws TException {
         final User sw360User = restControllerHelper.getSw360UserFromAuthentication();
         final Release sw360Release = releaseService.getReleaseForUserById(id, sw360User);
-        final CollectionModel<EntityModel<AttachmentDTO>> resources = attachmentService.getAttachmentDTOResourcesFromList(sw360User, sw360Release.getAttachments(), Source.releaseId(sw360Release.getId()));
+        final CollectionModel<EntityModel<Attachment>> resources = attachmentService.getAttachmentResourcesFromList(sw360Release.getAttachments());
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/ReleaseController.java:493]: Please remove the previous method(getAttachmentDTOResourcesFromList) if it is no longer used.

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java [hunk 2]
```diff
@@ -189,6 +189,7 @@ public void before() throws TException, IOException {
         given(this.attachmentServiceMock.updateAttachment(any(), any(), any(), any())).willReturn(att2);
         given(this.sw360VendorService.getVendorById(any())).willReturn(new Vendor("TV", "Test Vendor", "http://testvendor.com"));
         given(this.attachmentServiceMock.isAttachmentExist(eq("1231231254"))).willReturn(true);
+        given(this.attachmentServiceMock.getAttachmentResourcesFromList(any())).willReturn(CollectionModel.of(attachmentResources));
```
Comment (rudra-superrr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java:192]: Use proper indentation.

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java [hunk 3]
```diff
@@ -806,27 +807,20 @@ public void should_document_update_release() throws Exception {
...
+                                subsectionWithPath("_embedded.sw360:attachments").description("An array of <<resources-attachment, Attachments resources>>"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]filename").description("The attachment filename"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]sha1").description("The attachment sha1 value"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]createdBy").description("The attachment createdby value"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]createdTeam").description("The attachment created team value"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]createdOn").description("The attachment created on value"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]checkStatus").description("The attachment checkStatus value"),
+                                subsectionWithPath("_embedded.sw360:attachments.[]createdOn").description("The attachment checkedComment value"),
```
Comment (rudra-superrr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java:823]: it should be []checkedComment according to the description.

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java [hunk 4]
```diff
@@ -806,27 +807,20 @@ public void should_document_update_release() throws Exception {
...
-                                subsectionWithPath("_embedded.sw360:attachmentDTOes.[]checkedComment").description("The attachment checkedComment"),
-                                subsectionWithPath("_embedded.sw360:attachmentDTOes.[]checkStatus").description("The attachment checkStatus"),
-                                subsectionWithPath("_embedded.sw360:attachmentDTOes.[]usageAttachment").description("The usages in project"),
-                                subsectionWithPath("_embedded.sw360:attachmentDTOes.[]usageAttachment.visible").description("The visible usages in project"),
-                                subsectionWithPath("_embedded.sw360:attachmentDTOes.[]usageAttachment.restricted").description("The restricted usages in project"),
-                                subsectionWithPath("_embedded.sw360:attachmentDTOes.[]usageAttachment.projectUsages").description("The name of project usages"),
+                                subsectionWithPath("_embedded.sw360:attachments").description("An array of <<resources-attachment, Attachments resources>>"),
```
Comment (rudra-superrr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ReleaseSpecTest.java:816]: Also add other response fields descriptions like - attachmentType, created comment as these two fields are also present in response structure.

**Comment by rudra-superrr** [comment]
Code looks good.

**Pull request merged**